### ISSUE TITLE
Failed to get TCP info getsockopt: not implemented

### ISSUE THREAD

**Issue by prestonvanloon** [libp2p/go-tcp-transport#106]
After updating [prysm](github.com/prysmaticlabs/prysm) to a recent version of libp2p, many users are reporting this log. It seems to have something to do with metrics and it is a cause for concern.

```
Failed to get TCP info: raw-control tcp 10.0.0.2:13000: getsockopt: not implemented
```

It seems to be coming from this line: https://github.com/libp2p/go-tcp-transport/blob/v0.2.4/metrics.go#L135

How can we disable this metrics collection?
What is this log about?

Tracking in Prysm here: https://github.com/prysmaticlabs/prysm/issues/9733

**Comment by prestonvanloon** [comment]
I noticed this seems to be a windows only issue, but this issue gives us a bit of concern. 
Metrics collection have additional overhead, complexity, and side effects / bugs.
How can we disable metrics collection entirely for go-tcp-transport?

**Comment by aschmahmann** [comment]
Closed by #93 which is in https://github.com/libp2p/go-tcp-transport/releases/tag/v0.2.8.

As for the more general "how do I disable metrics" question I'll defer to @marten-seemann and if you'd like you can open a new issue.

However, IIRC in this particular case the only way to even trigger this error message on Windows is if you explicitly called metrics collection which is something that your application would have had to do.

**Comment by marten-seemann** [comment]
Metrics collection can currently not be disabled, as for all other libp2p / IPFS systems. So far, I haven't seen any benchmark that showed a significant performance impact of metrics collection in our stack (but I can be convinced otherwise by data).
